src/util/file.cpp

```diff
@@ -1,5 +1,74 @@
 #include "util/file.h"
 
+#include <QFileDialog>
+#include <QRegularExpression>
+
+namespace {
+
+const QRegularExpression kExtractExtensionRegex(R"(\(\*\.(.*)\)$)");
+
+} //anonymous namespace
+
+QString filePathWithSelectedExtension(const QString& fileLocationInput,
+        const QString& fileFilter) {
+    if (fileLocationInput.isEmpty()) {
+        return {};
+    }
+    QString fileLocation = fileLocationInput;
+    if (fileFilter.isEmpty()) {
+        return fileLocation;
+    }
+
+    // Extract 'ext' from QFileDialog file filter string 'Funky type (*.ext)'
+    const auto extMatch = kExtractExtensionRegex.match(fileFilter);
+    const QString ext = extMatch.captured(1);
+    if (ext.isNull()) {
+        return fileLocation;
+    }
+    const QFileInfo fileName(fileLocation);
+    if (!ext.isEmpty() && fileName.suffix() != ext) {
+        fileLocation.append(".").append(ext);
+    }
+    return fileLocation;
+}
+
+QString checkedFilePathWithSelectedExtension(const QString& caption,
```

Comment:
The function name sounds as if it will just do the check. It also pops up a dialog. 
This should be expressed by the name. 
Something like:
SaveFileDialogWithVerifiedExtension() or such

Reply:
yes, thank you. consider this WIP and the name will be adjusted.
What I care about here is whether this approach works for everyone, or not.

if this get's sufficient :+1: I'll implement it in other places, too, and change the name.

Reply:
considering the return value I'd name it `filePathWithVerifiedExtensionFromFileDialog`

Reply:
Thank you.

Reply:
Try to name functions starting with an imperative verb if possible. Like good commit messages ;)

Reply:
**get**FilePathWithVerifiedExtensionFromFileDialog ;)